tsl/test/expected/cagg_errors.out

```diff
@@ -158,33 +129,14 @@ from conditions
  WITH NO DATA;
 ERROR:  invalid continuous aggregate query
 DETAIL:  Window functions are not supported by continuous aggregates.
---aggregate without combine function
+--aggregate without combine function but stable function
 CREATE MATERIALIZED VIEW mat_m1 WITH ( timescaledb.continuous)
 AS
 Select json_agg(location)
 from conditions
  group by time_bucket('1week', timec) , location WITH NO DATA;
-ERROR:  aggregates which are not parallelizable are not supported
-;
-CREATE MATERIALIZED VIEW mat_m1 WITH ( timescaledb.continuous)
-AS
-Select sum(humidity), avg(temperature), array_agg(location)
-from conditions
- group by time_bucket('1week', timec) , location WITH NO DATA;
-ERROR:  aggregates which are not parallelizable are not supported
-;
--- userdefined aggregate without combine function
-CREATE AGGREGATE newavg (
-   sfunc = int4_avg_accum, basetype = int4, stype = _int8,
-   finalfunc = int8_avg,
-   initcond1 = '{0,0}'
-);
-CREATE MATERIALIZED VIEW mat_m1 WITH ( timescaledb.continuous)
-AS
-Select sum(humidity), newavg(temperature::int4)
-from conditions
- group by time_bucket('1week', timec) , location WITH NO DATA;
-ERROR:  aggregates which are not parallelizable are not supported
+ERROR:  only immutable functions supported in continuous aggregate view
+HINT:  Make sure the function includes only immutable expressions, e.g., time_bucket('1 hour', time AT TIME ZONE 'GMT').
```

Comment:
Is this testing what we want it to test or is the time_bucket the problem?

Reply:
Maybe we can have a version at the top of the file that demonstrates that this basic view definition does work, ie just count(*) or the like in this file, so that we know the normal stuff is valid and the time_bucket is valid etc in this file?

Reply:
> Is this testing what we want it to test or is the time_bucket the problem?

This is exact what we want to test... `json_agg` is stable:

```sql
fabrizio=# \df+ json_agg
List of functions
-[ RECORD 1 ]-------+--------------------------
Schema              | pg_catalog
Name                | json_agg
Result data type    | json
Argument data types | anyelement
Type                | agg
Volatility          | stable
Parallel            | safe
Owner               | fabrizio
Security            | invoker
Access privileges   | 
Language            | internal
Source code         | aggregate_dummy
Description         | aggregate input into json
```

Reply:
Ah. I think my confusion is because of the error message - we should, if possible, distinguish between stable things in the group by ie `time_bucket` and stable things as aggregates, so we can have different error messages for each so as not to confuse users. Or we could add a line to the error message with another example so folks aren't drawn only to the time_bucket. 

I think my problem is really with the hint.

HINT:  Make sure the function includes only immutable expressions, e.g., time_bucket('1 hour', time AT TIME ZONE 'GMT').

alternative: Make sure the view definition contains only IMMUTABLE expressions for aggregates and GROUP BYs.

Reply:
The current one makes it seem like I need to add a time zone to my time bucket and I think that's just confusing.
we might want to point them at the function volatility stuff. I just updated mine and I think it's pretty good? But I dunno.

Reply:
Or perhaps we could add a hint to run \df+ [function_name] in psql and look at the "volatility" field?

Reply:
> Or perhaps we could add a hint to run \df+ [function_name] in psql and look at the "volatility" field?

I think is not good show psql commands hints for users because they can use another tool to connect to postgres

Reply:
> alternative: Make sure the view definition contains no functions marked STABLE or VOLATILE in aggregates and GROUP BYs. Note that functions may be IMMUTABLE for one data type, but only STABLE for another.

@davidkohn88 what about?

Make sure the view definition includes only IMMUTABLE expressions. Note that functions or expressions may be IMMUTABLE for one data type, but only STABLE for another.

Reply:
Slight revision: Make sure the continuous aggregate definition includes only IMMUTABLE expressions. Note that functions or expressions may be IMMUTABLE for one data type, but STABLE  or VOLATILE for another.

Reply:
I will say that the first sentence feels redundant given the error message: only immutable functions supported in continuous aggregate view


Maybe something like: 
ERROR:  only immutable functions supported in continuous aggregate view
HINT: Make sure all functions in the continuous aggregate definition have IMMUTABLE volatility. Note that functions or expressions may be IMMUTABLE for one data type, but STABLE or VOLATILE for another.

That at least points them on how to look up the docs for volatility?

Reply:
Done!